### ISSUE TITLE
How can I call Tensorflow model.predict method? 

### ISSUE THREAD

**Issue by Tess314** [google/mediapipe#4930]
### Have I written custom code (as opposed to using a stock example script provided in MediaPipe)

None

### OS Platform and Distribution

Windows 10

### Python Version

3.10

### Task name (e.g. Image classification, Gesture recognition etc.)

Image classification

### Describe the actual behavior

Inability to use tf model.predict method

### Describe the expected behaviour

Ability to call model.predict()

### Standalone code/steps you may have used to try to get what you need

```shell
model.predict(test_data)
```

**Comment by kuaashish** [comment]
Hi @joezoug,

Could you provide any guidance or insights regarding this issue? Thank you

**Comment by Tess314** [comment]
(I am wanting to use this to generate a list of true and predicted labels for test data) if there is any alternative way already to do this please let me know, thanks.

**Comment by joezoug** [comment]
@Tess314 ,

The underlying Tensorflow model is in the instance variable `_model`, so you should be able to call predict like: `image_classifier_model._model.predict(*)`. See the [_create_model](https://github.com/google/mediapipe/blob/master/mediapipe/model_maker/python/vision/image_classifier/image_classifier.py#L123-L140) method of the ImageClassifier class for details on how this underlying Tensorflow model is created.

**Comment by Tess314** [comment]
I am getting the following error. Any advice?
![image](https://github.com/google/mediapipe/assets/71218139/4d9651e8-1479-4241-9a82-a465abebada8)

**Comment by joezoug** [comment]
Hi @Tess314,

TF model.predict function expects a dataset of type tf.data ([doc](https://www.tensorflow.org/api_docs/python/tf/keras/Model#predict)), but it seems like you are using a test_data which is an image_classifier.dataset.Dataset class. You will need to convert your image_classifier.dataset.Dataset object into a tf.data.Dataset object to use with the predict method. There is method called [`gen_tf_dataset`](https://github.com/google/mediapipe/blob/master/mediapipe/model_maker/python/core/data/dataset.py#L68C16-L68C16) which can do this for you. Here is an example of how it is used to call a similar TF model.fit function: https://github.com/google/mediapipe/blob/master/mediapipe/model_maker/python/core/tasks/classifier.py#L78-L115

**Comment by joezoug** [comment]
Hi @Tess314,

I believe you need to change your code from ```image_classifier_model._model.predict(test_data)```
to this line:
```image_classifier_model._model.predict(test_data.gen_tf_dataset())```.

Let me know if that works!

**Comment by Tess314** [comment]
That words, thank you!
One more question:
How can I iterate over type mediapipe_model_maker.python.vision.image_classifier.dataset.Dataset and print each element?

**Comment by joezoug** [comment]
You will need to convert to tf.data using the `gen_tf_dataset` method.

You can do something like this:
```
for elem in test_data.gen_tf_dataset():
    print(elem)
```
